### ISSUE TITLE
The "object queue" and no delay (dehoo)

### ISSUE THREAD

**Issue by captdeaf** [pennmush/pennmush#1222]
For a bit of background information to ensure everybody's on the same page:

We have, two queues: "Player Queue" and "Object Queue". Whenever anything is in the "player queue", it runs, almost immediately.

What goes in the player queue: Any `@trigger`, `@wait`, etc with an executor OR enactor that is a player.
What goes in object queue: Anything w/ an executor and enactor BOTH being an object.

Anything in an object queue is deliberately delayed until the next second. This isn't a 1 second delay, but quite literally "the next second". Meaning if it's now 123seconds and 450, then depending on activity, it will be dumped into the object queue at any time between 550 ms and 1s550ms from now. (Assuming my latest millisecond patch - prior, the uncertainty was larger)

Practically, what does this mean?

Given this code (`me` is a player), can you predict the output, and its timing?
```
@create Flip
&FLIP flip=say [secs()] flip1 ; @trigger me/flip2
&FLIP2 flip=say [secs()] flip2 ; @trigger me/flip3
&FLIP3 flip=say [secs()] flip3
&ZIP me=say [secs()] zip1 ; @trigger me/zip2
&ZIP2 me=say [secs()] zip2 ; @trigger me/zip3
&ZIP3 me=say [secs()] zip3

&example me=@trigger me/zip ; @trigger flip/flip
```

<details><summary>View result</summary><p>

```
Walker - Triggered.
Walker - Triggered.
flip - Triggered.
You say, "1534281318 zip1"
Walker - Triggered.
flip says, "1534281318 flip1"
You say, "1534281318 zip2"
Walker - Triggered.
You say, "1534281318 zip3"
flip says, "1534281319 flip2"
flip says, "1534281320 flip3"
```
</p></details>

Related:

> <Hardcode> Qon says, "Only time I've perceived a slowness was always due to some stupid 
usage of @trigger in a global command"

This full second wait seems archaic. I think we can do away with it.

Two options that come to mind (I welcome more!):

1) Don't move object queue, instead give it same priority as player queue and empty player queue then object queue. This would keep (roughly) the same ordering we already have. The downside: If player_queue never empties, then object_queue will never be run.

2) Eliminate object queue, only use a single queue. Downside: A bunch of greedy objects can hang player_queue.`**`

Also, to be clear: Incoming commands from sockets are processed immediately, they never make it to the player_queue`***`. I'd like to say they take priority over even player_queue, except any time network data comes in, `config(active_queue_chunk)` queue entries are executed. (I'm open to moving this to post-socket-commands).

`**`Another thing I'd like to toss in as an idea: Make socket commands (and only socket commands) not queue $-commands, but run them immediately inline. To make Penn seem a wee bit faster for players, _and_ it obviates the player_queue.

`***`: incoming hardcode and ??/inline commands from socket are run immediately. $-commands are matched immediately, but then placed on the queue for execution.

My own vote is to eliminate the double queue system, and to make socket input effectively /inline $-commands.

Thoughts?

**Comment by mike347** [comment]
I like the sound of this change -- Making socket commands run immediately (inline), and go to a 1-queue system. 

My only concern is what are the downsides of this, are there any besides the ability for 'greedy objects' to hog the queue? Can the fallout be limited, by giving a (slight) priority to player queued commands when the queue is getting backed up?

**Comment by talvo** [comment]
Stuff you enter directly from a client isn't queued anyway, which means the only time "player queued" currently comes into effect, really, is when you're running an $-command (whose action list is then queued). If that gets run inline, there's basically no player queueing going on. (You could kinda consider an $-command which then does an `@dolist` to be player queueing, but atm I think that winds up on the object queue anyway, so it wouldn't have any effect on it.)

**Comment by captdeaf** [comment]
Possibly for now we can detect when, say, the mush goes 30 seconds without an empty queue (after the startup queue pumps are complete). If that happens, be noisy about it? `@wizwall` ?. Then again, not a fan of that in case of, e.g: `@dol lsearch()=@@updatealltheroomsorsomething` which may fill the queue for a longish period of time on large mushes.

**Comment by captdeaf** [comment]
On the other hand, I may be worried about something that isn't actually the case for enough mushes to make a difference. Any mush using a variant on `&foo me=... ; @trigger me/foo` can fix their dang code :D.

**Comment by captdeaf** [comment]
Closed by PR #1224
